TombEngine/Objects/TR5/Entity/tr5_roman_statue.cpp

```diff
@@ -306,37 +278,47 @@ namespace TEN::Entities::Creatures::TR5
 		short joint0 = 0;
 		short joint1 = 0;
 		short joint2 = 0;
+		unsigned int jointBit = 0;
 
 		auto* item = &g_Level.Items[itemNumber];
 		auto* creature = GetCreatureInfo(item);
 
 		auto prevMeshSwapBits = item->Model.MeshIndex;
 
 		// At determined HP values, roman statues sheds material.
-		if (item->HitPoints < 1 && !item->TestMeshSwapFlags(0x10000))
+		if (item->HitPoints < 1 && !(item->ItemFlags[1] & (MS_HEAVY_DMG >> 4)))
 		{
 			ExplodeItemNode(item, 16, 0, 8);
-			item->MeshBits |= 0x10000;
-			item->SetMeshSwapFlags(0x10000);
+			jointBit =  MS_HEAVY_DMG;
+			item->MeshBits |= jointBit;
+			item->ItemFlags[1] |= jointBit >> 4;
+			jointBit |= jointBit;
+			item->SetMeshSwapFlags(jointBit);		
 		}
-		else if (item->HitPoints < 75 && !item->TestMeshSwapFlags(0x100))
+		else if (item->HitPoints < 75 && !(item->ItemFlags[1] & MS_HARD_DMG))
 		{
 			ExplodeItemNode(item, 8, 0, 8);
-			item->MeshBits |= 0x100;
-			item->SetMeshSwapFlags(0x100);
+			jointBit = MS_HARD_DMG;
+			item->MeshBits |= jointBit;
+			item->ItemFlags[1] |= jointBit;
+			item->SetMeshSwapFlags(item->ItemFlags[1]);		
 		}
-		else if (item->HitPoints < 150 && !item->TestMeshSwapFlags(0x400))
```

Comment:
I am not sure why `TestMeshSwapFlags` was reverted here and further. It's basically the same test?

Reply:
changed